Assets/Capstone/Scripts/Environmental/Sun.cs

```diff
@@ -46,69 +67,149 @@ void Update()
+            if (stars.activeSelf)
             {
-                dayCounter++;
-
-                //if dayCounter is multiple of 3, randomize wind
-                if(dayCounter % 3 == 0)
-                {
-                    RandomizeWinds();
-                }
+                stars.SetActive(false);
             }
+
+            //lighting
+            LerpLighting(morn, ambientMorn, skyMorn);
+            //time bool
+            SwitchTimeState(0);
+
+            //resets day bools and sleep stuff every morning
+            NewDay();
         }
-        else if(transform.position.x < middayInterval && transform.position.x > duskInterval)
+        //is Midday
+        else if (angleInDegrees > middayInterval && angleInDegrees < duskInterval)
         {
-            sun.color = Color.Lerp(sun.color, dusk, Time.deltaTime / 10);
-            RenderSettings.ambientLight = Color.Lerp(RenderSettings.ambientLight, ambientDusk, Time.deltaTime / 10);
-            //sun.intensity = Mathf.Lerp (sun.intensity, 2, Time.deltaTime);
-            isMidday = true;
-            isMorning = false;
-            isDusk = false;
-            isNight = false;
+            //lighting
+            LerpLighting(mid, ambientMid, skyMid);
+            //time bool
+            SwitchTimeState(1);
         }
-        else if (transform.position.x < duskInterval && transform.position.x > nightInterval)
+        //is Dusk
+        else if (angleInDegrees > duskInterval && angleInDegrees < nightInterval)
         {
-            sun.color = Color.Lerp(sun.color, night, Time.deltaTime / 10);
-            RenderSettings.ambientLight = Color.Lerp(RenderSettings.ambientLight, ambientNight, Time.deltaTime / 10);
-            //sun.intensity = Mathf.Lerp (sun.intensity, 2, Time.deltaTime);
-            isMidday = false;
-            isMorning = false;
-            isDusk = true;
-            isNight = false;
+            //lighting
+            LerpLighting(dusk, ambientDusk, skyDusk);
+            //time bool
+            SwitchTimeState(2);
         }
-        else if (transform.position.x < nightInterval)
+        //is Night
+        else if (angleInDegrees > nightInterval && angleInDegrees < morningInterval)
         {
-            isMorning = false;
-            isMidday = false;
-            isDusk = false;
-            isNight = true;
-            //waterDay.SetActive(false);
-            //waterNight.SetActive(true);
-            
+            //lighting
+            LerpLighting(night, ambientNight, skyNight);
+            //time bool
+            SwitchTimeState(3);
+
             //when its night, yesterday catches up to dayCounter
             yesterday = dayCounter;
+
+            //turn on stars
+            if (!stars.activeSelf )
+            {
+                stars.SetActive(true);
+            }
+        }
+    }
+
+    //pass the state you want to set true
+    void SwitchTimeState(int state)
+    {
+        switch (state)
+        {

```

Comment:
With the enums you won't need so many lines of code to represent what you want to do.